platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/client/impl/SecureCxfClientFactoryImpl.java

```diff
@@ -358,20 +367,30 @@ public WebClient getWebClient() {
    * each new request in order to ensure that the security token is up-to-date each time.
    */
   public T getClientForSubject(Subject subject) {
-    String asciiString = clientFactory.getAddress();
-
-    T newClient = getNewClient();
+    final java.lang.SecurityManager security = System.getSecurityManager();
 
-    if (!basicAuth && StringUtils.startsWithIgnoreCase(asciiString, "https")) {
-      if (subject instanceof ddf.security.Subject) {
-        RestSecurity.setSubjectOnClient(
-            (ddf.security.Subject) subject, WebClient.client(newClient));
-      }
+    if (security != null) {
+      security.checkPermission(CREATE_CLIENT_PERMISSION);
```

Comment:
This just makes me happy.

Reply:
Yeah, I'm not denying the permission anywhere yet, but it is ready!